lib/list_versions.go

```diff
@@ -14,15 +14,7 @@ type tfVersionList struct {
 	tflist []string
 }
 
-// getTFList :  Get the list of available terraform version given the hashicorp url
-func getTFList(mirrorURL string, preRelease bool) ([]string, error) {
-	logger.Debugf("Get list of terraform versions")
-	result, err := getTFURLBody(mirrorURL)
-	if err != nil {
-		return nil, err
-	}
-
-	var tfVersionList tfVersionList
+func getVersionsFromBody(body string, preRelease bool, tfVersionList *tfVersionList) {
```

Comment:
Just a wild suggestion: would it make sense to employ a HTML parsing lib (e.g. `net/html` or similar) to parse HTML? 🤔 
Might probably be a scope for standalone PR though.

Reply:
From the discussion in #434, I'd assume (hope) that maybe we'd look at moving to the JSON endpoints for Terraform/OpenTofu (if we create a PR there) and wouldn't have a need for it.
I guess using a HML parser would also mean knowing the exact layout of the page and traversing it accordingly - as brittle as using regex on HTML to obtain the versions is, I would imagine HTML parsing it _could_ be worse in this sense?

Reply:
Your point makes sense to me. In spite of moving to JSON there's no point in re-factoring to HTML parsing using lib 🤝 
As for the parsing using regex vs parsing using purpose-built lib, I'd say why don't we parse JSON using regexp then? (it is possible. I tried it and it worked) =))